File: src/main/java/me/kikugie/techutils/utils/ResponseMuffler.java
package me.kikugie.techutils.utils;

import java.util.ArrayDeque;

public class ResponseMuffler {
    private static final ArrayDeque<String> muteQueue = new ArrayDeque<>();

    public static void scheduleMute(String matcher) {
        muteQueue.add(matcher);
    }

    public static void pop() {
        muteQueue.remove();
    }

    public static void clear() {
        muteQueue.clear();
    }

    public static boolean matches(String message) {
        if (muteQueue.isEmpty()) return false;
        return message.matches(muteQueue.peek());
    }
}

File: src/main/kotlin/me/kikugie/techutils/utils/ResponseMuffler.kt
package me.kikugie.techutils.utils

import java.util.*

object ResponseMuffler {
    private val muteQueue = ArrayDeque<String>()
    fun scheduleMute(matcher: String) {
        muteQueue.add(matcher)
    }

    fun pop() {
        muteQueue.remove()
    }

    fun clear() {
        muteQueue.clear()
    }

    fun matches(message: String): Boolean {
        return if (muteQueue.isEmpty()) false else message.matches(muteQueue.peek().toRegex())
    }
}